libusb/os/windows_winusb.c

```diff
@@ -3067,8 +3067,41 @@ static void WINAPI winusbx_native_iso_transfer_continue_stream_callback(struct l
 	} else {
 		// If the transfer wasn't successful we reschedule the transfer while forcing it
 		// not to continue the stream. This might results in a 5-ms delay.
-		transfer_priv->iso_break_stream = TRUE;
-		libusb_submit_transfer(transfer);
+		libusb_transfer_cb_fn postSubmissionHandler = NULL;
+		bool resubmitTransfer= TRUE;
+
+		if (transfer->callback) {
+			// Notify the user callback that a resubmission is about to take place
+			// This gives it an option to interfer or schedule submission of other transfers after this one
+			transfer->status = LIBUSB_TRANSFER_STALL;
+			transfer->callback(transfer);
+			if (transfer->callback != transfer_priv->iso_user_callback) {
+				// Callback temporarily used as signal by user callback
```

Comment:
I like the idea of having extra steps when handling ISO transfers, and the algorithm presented here completely make sense, but I find the implementation to be a little too "hacky" to be merge as is, mainly because of 2 things: i) reusing LIBUSB_TRANSFER_STALL value and ii) temporary reusing transfer->callback to inject another post-submission callback.
I would suggest to:
a) Introduce a new value to the libusb_transfer_status enum, such as LIBUSB_TRANSFER_RESUBMISSION. I understand that STALL value is not currently use for ISO, so it will work as proposed, but having a new value would make thing clearer and would simplify writing documentation about this new feature, instead of intricated sentences in the LIBUSB_TRANSFER_STALL section.

b)  Adding a new post_submission_callback field to libusb_transfer structure. After all we already have ISO specific fields that are not used for other type of transfer. I think it would make things clearer, less error prone and again maybe ease documentation writing. 

I think those 2 changes would pay in the long run, better integrating the feature, "auto-documenting" it and making it easier to comprehend for people that don't look into the code for implementation details